spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1808,40 +1813,44 @@ trait OptimisticTransactionImpl extends TransactionalWrite
       case _ =>
     }
     collectAutoOptimizeStats(numAdd, numRemove, actions.iterator)
-    val info = currentTransactionInfo.commitInfo
-      .map(_.copy(readVersion = None, isolationLevel = None)).orNull
     setNeedsCheckpoint(attemptVersion, postCommitSnapshot)
-    val stats = CommitStats(
-      startVersion = snapshot.version,
-      commitVersion = attemptVersion,
-      readVersion = postCommitSnapshot.version,
-      txnDurationMs = NANOSECONDS.toMillis(commitEndNano - txnStartNano),
-      commitDurationMs = NANOSECONDS.toMillis(commitEndNano - commitStartNano),
-      fsWriteDurationMs = NANOSECONDS.toMillis(commitEndNano - fsWriteStartNano),
-      stateReconstructionDurationMs =
-        NANOSECONDS.toMillis(postCommitReconstructionTime - commitEndNano),
-      numAdd = numAdd,
-      numRemove = numRemove,
-      numSetTransaction = numSetTransaction,
-      bytesNew = bytesNew,
-      numFilesTotal = postCommitSnapshot.numOfFiles,
-      sizeInBytesTotal = postCommitSnapshot.sizeInBytes,
-      numCdcFiles = numCdcFiles,
-      cdcBytesNew = cdcBytesNew,
-      protocol = postCommitSnapshot.protocol,
-      commitSizeBytes = jsonActions.map(_.size).sum,
-      checkpointSizeBytes = postCommitSnapshot.checkpointSizeInBytes(),
-      totalCommitsSizeSinceLastCheckpoint = postCommitSnapshot.deltaFileSizeInBytes(),
-      checkpointAttempt = needsCheckpoint,
-      info = info,
-      newMetadata = newMetadata,
-      numAbsolutePathsInAdd = numAbsolutePaths,
-      numDistinctPartitionsInAdd = distinctPartitions.size,
-      numPartitionColumnsInTable = postCommitSnapshot.metadata.partitionColumns.size,
-      isolationLevel = isolationLevel.toString,
-      numOfDomainMetadatas = numOfDomainMetadatas,
-      txnId = Some(txnId))
-    recordDeltaEvent(deltaLog, DeltaLogging.DELTA_COMMIT_STATS_OPTYPE, data = stats)
+
+    if (spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_COLLECT_COMMIT_STATS)) {
```

Comment:
When false can't we record basic info that doesn't require expensive extra computation?

Reply:
Hi @felipepessoto yeah that's a good idea.  I will amend this PR to add your suggestion.

Reply:
Hi @felipepessoto I amended the PR so now it only skips logging the expensive fields.

(I also removed the part of this PR which was about skipping calculating the row ID high watermark.  That one is a slightly different issue, because it forces the computation of the old snapshot, not the new snapshot.  I still think there is an issue there to be solved, but that can be tackled separately)

Does it make sense to you what I am trying to achieve here?  Have I explained OK why it is a big problem to compute the SnapshotState during each transaction?  There is an opportunity here to make Delta much much faster for streaming writes.

Reply:
Makes sense to me. Thanks. We need some of the maintainers to review it.

Reply:
@istreeter just curious about CREATE TABLE scenario, have you measured the diff with your optimization? Creating an empty table takes several seconds, I hope this also improves it.

Reply:
Hi @felipepessoto yes there is an improvement.  I tested with a remote bucket in S3:

```
spark.sql("""CREATE TABLE delta.`s3a://<redacted>/empty` (id string) USING DELTA """)
```

Before this PR, the job takes *9.4 seconds* and I can see from the logs the snapshot state is computed.
After this PR, the job takes *6.4 seconds*, and it skips computing the snapshot state.

I repeated this test several times, and the timings were fairly consistent.